### ISSUE TITLE
Private fields only used in nullish coalescing assignment are incorrectly seen by the compiler as being never read

### ISSUE THREAD

**Issue by dmo-odoo** [microsoft/TypeScript#52720]
# Bug Report

### 🔎 Search Terms

private fields never read nullish coalescing assignment

### 🕗 Version & Regression Information

This does not look to be a regression as the earlier version of Typescript from the Playground that supports both private fields _and_ nullish coalescing assigment (that would be 4.0.5) already exhibits this behavior.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noUnusedLocals=true&target=99&ts=4.9.5#code/MYGwhgzhAEAa0G8BQ1XQMQAcCmA7AJgJa4DmAXNAAoBOA9gLaETYA8AbrYfgHzQA+0AK4FsAM2LZ8AbmgB6WdADkWPEVKLoTaPmygw1SdABGggC6bTMNmBCDsmmLmxts1aAbD4AdAAoAbACMAMxBAJQoaPi0AMoM2KYAFsQk0QCeuMAJdLi0ghAgqQCCBJTguISiqbH08UmkAKIgzIUQ6ZnZufmpPqGIEWhowLS4ELQg2F4gtCQ+igAqSTCY+uZarRleiqFS-QPQiUxeKgTJ0AD8ZwC8VHSMzF4GoyAuPV6JeD490Je8yHv-+0WRxwJ1I3yEInETmkuwB0CGIzGEymM3miwc0EgbUxBHhNnG+GgwwKRIy9lEtDc+MB9mA+JgploNOgNUStEJxGZEDANRZhGAdFMkAA1vt+cLNuE4QBfba7aVIaVAA)

### 💻 Code

```ts
class X {
    #pending: Promise<void> | undefined; // '#pending' is declared but its value is never read.(6133)
    doSomethingSynchronouslyAndPlanifySomethingElseAsynchronously() {
        console.log('This part is sync.');
        this.#pending ??= Promise.resolve().then(() => {
            this.#pending = undefined;
            console.log('This part is async and called only once for all the calls to the method during the same microtask tick.')
        });
    }
}
```

### 🙁 Actual behavior

Private fields that are only used in nullish coalescing assignments are incorrectly seen by the compiler as being never read under `noUnusedLocals`.

### 🙂 Expected behavior

The nullish coalescing operator should correctly count for a "read" operation. Interestingly, it does so if the field is _not_ private. The above code with a standard public field instead of a private one correctly does _not_ raise a compiler error under `noUnusedLocals`.

**Comment by fatcerberus** [comment]
Making the field public actually bypasses this check entirely (because then *someone else* might want to read it), fwiw.

In most cases of compound assignment (e.g. `+=`) this error would be correct; the value is never read, except by the assignment operation itself.  The intent of the error is that you have a variable which is assigned to but whose value never affects anything else (except its own value); null coalescence is unique though in that `??=` is short-circuiting, so the value is actually observable.

**Comment by MartinJohns** [comment]
Duplicate of #44802.
